test/e2e/performanceprofile/functests/1_performance/performance.go

```diff
@@ -1018,6 +1019,52 @@ var _ = Describe("[rfe_id:27368][performance]", func() {
 			})
 		})
 	})
+
+	Context("Offlined CPU API", func() {
+		It("should set offline cpus after deploy PAO", func() {
+			By("Backing up the profile")
+			initialProfile := profile.DeepCopy()
+
+			// Create new performance with offlined
+			reserved := performancev2.CPUSet("0")
+			isolated := performancev2.CPUSet("1")
+			offlined := performancev2.CPUSet("2-3")
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Enable UserLevelNetworking and add Devices in Profile")
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reserved,
+				Isolated: &isolated,
+				Offlined: &offlined,
+			}
+			By("Updating the performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By("Waiting for MCP being updated")
+			mcps.WaitForCondition("worker-cnf", mcv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+			//Check offlined cpus are setting correctly
+			for _, node := range workerRTNodes {
+				Eventually(func() bool {
+					offlinedOutput, err := nodes.ExecCommandOnMachineConfigDaemon(&node, []string{"cat", "/sys/devices/system/cpu/offline"})
+					Expect(err).ToNot(HaveOccurred())
+					fmt.Printf("mario %s", offlinedOutput)
+					return strings.Contains(string(offlinedOutput), string(offlined))
+				}, cluster.ComputeTestTimeout(15*time.Minute, RunningOnSingleNode), 1*time.Minute).Should(BeTrue(), "could not get correct offline cpus")
+			}
+
+			By("Reverting the Profile")
+			spec, err := json.Marshal(initialProfile.Spec)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testclient.Client.Patch(context.TODO(), profile,
+				client.RawPatch(
+					types.JSONPatchType,
+					[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+				),
+			)).ToNot(HaveOccurred())
```

Comment:
we need to wait for the revert to happen before to call the test done, and we need to make sure we revert is attempted anyway even if the test exits early

Reply:
done